packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+
+const double _kMinFlingVelocity = 1.0;  // screen width per second
+
+// Used for iOS.
+class CupertinoPageTransition extends AnimatedWidget {
+  static final FractionalOffsetTween _kTween = new FractionalOffsetTween(
+    begin: FractionalOffset.topRight,
+    end: -FractionalOffset.topRight
+  );
+
+  CupertinoPageTransition({
+    Key key,
+    Animation<double> animation,
+    this.child
+  }) : super(
+    key: key,
+    listenable: _kTween.animate(new CurvedAnimation(
+      parent: animation,
+      curve: new _CupertinoTransitionCurve(null)
+    )
+  ));
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    // TODO(ianh): tell the transform to be un-transformed for hit testing
+    // but not while being controlled by a gesture.
+    return new SlideTransition(
+      position: listenable,
+      child: new PhysicalModel(
+        shape: BoxShape.rectangle,
+        color: const Color(0x00000000), // we don't need the canvas background, just the elevation
```

Comment:
We _are_ filling it with an arbitrary colour. :-)

I think if we want the background to be gray, we should set it to gray. I think, though, that iOS probably has an opinion of what the default colour should be here, probably some off-white, and we should use that. Consider if someone's route is just a `Text` widget. It would look weird to have a shadow, but not have a background. If we want a shadow, we should commit to a background, IMHO.

cc @jason-simmons who may be able to comment on the PhysicalModel API requiring a colour.
cc @abarth who may have opinions as well.